neuraxle/checkpoints.py

```diff
@@ -19,185 +19,442 @@
+        with open(self.get_checkpoint_file_path(checkpoint_path, current_id), 'wb') as file:
+            pickle.dump((current_id, data_input, expected_output), file)
 
-        for current_id, data_input, expected_output in data_container:
-            with open(self.get_checkpoint_file_path(current_id), 'rb') as file:
-                (checkpoint_current_id, checkpoint_data_input, checkpoint_expected_output) = \
-                    pickle.load(file)
-                list_data_container.append(
-                    current_id=checkpoint_current_id,
-                    data_input=checkpoint_data_input,
-                    expected_output=checkpoint_expected_output
-                )
+        return True
 
-        return list_data_container
+    def read_current_id_checkpoint(self, checkpoint_path: str, current_id) -> Tuple[str, Iterable, Iterable]:
+        """
+        Read the data inputs, and expected outputs for the given current id using pickle.load.
 
-    def save_checkpoint(self, data_container: DataContainer) -> DataContainer:
+        :param checkpoint_path:
+        :param current_id:
+        :return: tuple(current_id, checkpoint_data_input, checkpoint_expected_output)
+        :rtype: tuple(str, Iterable, Iterable)
         """
-        Save pickle files for data inputs and expected output to create a checkpoint
+        with open(self.get_checkpoint_file_path(checkpoint_path, current_id), 'rb') as file:
+            (checkpoint_current_id, checkpoint_data_input, checkpoint_expected_output) = \
+                pickle.load(file)
+            return checkpoint_current_id, checkpoint_data_input, checkpoint_expected_output
 
-        :param data_container: data to resume
+    def get_checkpoint_file_path(self, checkpoint_path: str, current_id) -> str:
+        """
+        Get the checkpoint file path for a data input id
+
+        :param checkpoint_path:
+        :param current_id:
+        :return: path
+        :rtype: str
+        """
+        return os.path.join(
+            checkpoint_path,
+            '{0}_{1}.pickle'.format(current_id, self.checkpoint_file_name_suffix)
+        )
+
+
+class DataContainerCheckpointer(BaseCheckpointer):
+    """
+    Class used to checkpoint a data container for it's given execution mode.
+    This class uses a BaseDataContainerSaver to actually save the data checkpoint.
+
+    DataContainerCheckpointer loops through a data container to save, or load the data inputs,
+    and expected outputs using the current_ids inside the data container.
+    """
+    def __init__(
+            self,
+            data_container_saver: BaseDataContainerSaver,
+            execution_mode: ExecutionMode
+    ):
+        BaseCheckpointer.__init__(self, execution_mode)
+
+        self.execution_mode: ExecutionMode = execution_mode
+        self.data_checkpointer: BaseDataContainerSaver = data_container_saver
+
+    @staticmethod
+    def create_pickle_checkpointer(execution_mode: ExecutionMode):
```

Comment:
This is strange considering the name of the class doesn't talk of pickles. I didn't really read the code so don't be mad for this comment, it just feels strange. It's like if a high-level class suddenly gets responsible for naming the tech it uses, coupling the tech it uses with the concept (e.g.: abstract repo with concrete implementation). Would the abstract repo has a method with a pickle in its name? Idk I didn't understand the code here really, this is just a feeling I get and it might be wrong.

Reply:
Should this class be named PickleDataContainerCheckpointer ?

Reply:
you're right I was high or something that doesn't feel right. I like your suggestion